tests/integration/elder.test.ts

```diff
@@ -0,0 +1,79 @@
+import Elder, {
+  Model,
+  StringType,
+  type,
+  Adapter,
+  PostgresAdapter,
+  Collection
+} from '../../src'
+
+const config = { adapters: { default: { database: 'ash' } } }
+
+describe('Elder', () => {
+  test('model setup', async () => {
+    class CatModel extends Model {
+      @type('string') name: string
+    }
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel }
+    })
+    const cats = await CatModel.all()
+    expect(cats).toBeInstanceOf(Collection)
+    expect(cats[0]).toBeInstanceOf(CatModel)
+    expect(cats[0].name).toBe('fluffy')
+    return orm.destroy()
+  })
+
+  test('model with custom type', async () => {
+    class CatModel extends Model {
+      @type('cat-name') name: string
+    }
+    class CatName extends StringType {}
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel },
+      types: { 'cat-name': CatName }
+    })
+    const cats = await CatModel.all()
+    expect(cats[0].name).toBe('fluffy')
+    return orm.destroy()
+  })
+
+  test('model with custom scoped type', async () => {
+    class CatModel extends Model {
+      @type('name') name: string
+    }
+    class CatName extends StringType {}
+    const orm = Elder.create({
+      config,
+      models: { cat: CatModel },
+      types: { 'cat:name': CatName }
```

Comment:
Ahh yea, good call, not clear at all whats going on.
It's a scoped type. It can only be used by the model before the `:` So `cat:name` can only be used in the Cat model and is used like: `@type('name') name` ie. When using the decorator the `cat:` part is omitted.